chirpstack/src/storage/fields/big_decimal.rs

```diff
@@ -0,0 +1,93 @@
+use diesel::{
+    backend::Backend,
+    {deserialize, serialize},
+};
+#[cfg(feature = "postgres")]
+use diesel::{pg::Pg, sql_types::Numeric};
+#[cfg(feature = "sqlite")]
+use diesel::{sql_types::Double, sqlite::Sqlite};
+
+#[derive(Clone, Debug, Eq, PartialEq, AsExpression, FromSqlRow)]
+// Sqlite has no native arbitrary precision type so use double for now
+// TODO: Maybe use string representation instead?
```

Comment:
I think double for now is fine. Using string representation it becomes hard to perform any SQL query on the value.